app/src/main/java/com/doublesp/coherence/activities/MainActivity.java

```diff
@@ -1,109 +1,89 @@
+        IdeaCreationActionHandler.IdeaPreviewHandlerInterface,
+        ListFragmentActionHandler.IdeaShareHandlerInterface {
+
     public static final int RC_SIGN_IN = 1;
-    private static FirebaseRecyclerAdapter<UserList, ItemViewHolder> mFirebaseRecyclerAdapter;
-    private String mUsername;
+    static final String IDEA_PREVIEW_FRAGMENT = "IDEA_PREVIEW_FRAGMENT";
+    static final String LIST_COMPOSITION_FRAGMENT = "LIST_COMPOSITION_FRAGMENT";
+    ActivityMainBinding binding;
+    HomeActivitySubComponent mActivityComponent;
     private FirebaseDatabase mFirebaseDatabase;
-    private DatabaseReference mListsDatabaseReference;
     private FirebaseAuth mFirebaseAuth;
     private FirebaseAuth.AuthStateListener mAuthStateListener;
     private DatabaseReference mUsersDatabaseReference;
-    private RecyclerView mRecyclerView;
-    private FloatingActionButton mCustomList;
-    private FloatingActionButton mRecipeList;
-    private FloatingActionsMenu mFloatingActionsMenu;
+    private String mUsername;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
-
-        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
-        // TODO: need to define actions for custom list
-        mCustomList = (FloatingActionButton) findViewById(R.id.custom_list);
-        mRecipeList = (FloatingActionButton) findViewById(R.id.recipe_list);
-        mFloatingActionsMenu = (FloatingActionsMenu) findViewById(R.id.multiple_actions);
-        mRecipeList.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Intent intent = new Intent(MainActivity.this, HomeActivity.class);
-                startActivity(intent);
-            }
-        });
+        Fabric.with(this, new Crashlytics());
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
+        binding.viewpager.setAdapter(
+                new HomeFragmentPagerAdapter(getSupportFragmentManager(), MainActivity.this));
+        binding.tabs.setupWithViewPager(binding.viewpager);
+        CoherenceTabUtils.bindIcons(MainActivity.this, binding.viewpager, binding.tabs);
 
         mFirebaseDatabase = FirebaseDatabase.getInstance();
         mFirebaseAuth = FirebaseAuth.getInstance();
 
-        mListsDatabaseReference = mFirebaseDatabase.getReference().child(
-                ConstantsAndUtils.USER_LISTS).child(ConstantsAndUtils.getOwner(this));
         mUsersDatabaseReference = mFirebaseDatabase.getReference().child(ConstantsAndUtils.USERS);
 
-        final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
-        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
-        mRecyclerView.setLayoutManager(linearLayoutManager);
-
-        mFirebaseRecyclerAdapter = new FirebaseRecyclerAdapter<UserList, ItemViewHolder>(
```

Comment:
you have removed the code to populate the recycler view with saved lists, is that intentional because you are thinking of some changes on how we populate that data ? Currently if you remove this no list will be synced from the server and we will have only blank data.